### PR TITLE
WIP: Add role documentation

### PR DESCRIPTION
**Pull request by 14rcole**
Add documentation on how to develop Ansible Galaxy roles for LinchPin

### CODE REVIEW

**Comment by 14rcole** [comment]
Let me know if all of this reads well, there's enough examples, etc.  Still working on the `Testing` section but I think everything else is finished

**Comment by abraverm** [comment]
When I started developing a provider for Linchpin, these two things that would've helped me:
 - How to install a simple\example role (repo) in  a fresh installed Linchpin - where to put the files, what is nice to have and what is critical, what is the best practice, etc.
 - Explain how different parts of the role are being used during Linchpin execution: "If you run `linchpin up` with PinFile using the example role, it will start by doing X, and it will load the role Y file and look for Z..." 

Maybe we should ask outsiders (OASIS developers?)

@JunqiZhang0, you are relatively new to Linchpin, what do you think?

File: docs/source/developing_roles.rst [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+This resource definition would be valid under the schema:
+
+.. code-block:: yaml
+
+    - name: "master-node"
+      role: "dummy_node"
+      count: 3
+
+
+This one, however, would not.  Notice how "count" is a string here.  To learn more about cerberus, visit the link above.
```
Comment (samvarankashyap) [inline docs/source/developing_roles.rst]: @14rcole 
"This one " is ambiguous 
I would say, "The following one" instead

File: docs/source/gs/pinfile.rst [hunk 2]
```diff
@@ -37,6 +37,24 @@ A :term:`PinFile` takes a :term:`topology` and an optional :term:`layout`, among
...
+To use a PinFile with an Ansible Galaxy role, simply provide the role name as the `resource_group_type`.  An example is shown below.
+
+.. code-block:: yaml
+
+    dummy-new:
+      topology:
+        topology_name: "dummy_cluster" # topology name
+        resource_groups:
+          - resource_group_name: "dummy"
+            resource_group_type: "14rcole.ansible_role_lp_dummy"
```
Comment (abraverm) [inline docs/source/gs/pinfile.rst]: Which PR makes this work?

Comment (14rcole) [reply docs/source/gs/pinfile.rst]: I believe #1409

**Comment by samvarankashyap** [comment]
@14rcole I thought the galaxy docs are still work in progress. 
Can you create a new issue on github for the work remaining ( if any)?

**Comment by 14rcole** [comment]
The only thing I'd like to add is some more info on writing testinfra tests.  That we can add later, this at least allows users to develop roles.  I'll create an issue with the remaining work

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**